roles/kubernetes/control-plane/templates/admission-controls.yaml.j2

```diff
@@ -4,6 +4,6 @@ plugins:
 {% for plugin in kube_apiserver_enable_admission_plugins %}
 {% if plugin in kube_apiserver_admission_plugins_needs_configuration %}
 - name: {{ plugin }}
-  path: {{ kube_config_dir }}/{{ plugin | lower }}.yaml
+  path: {{ kube_config_dir }}/admission-controls/{{ item | lower }}.yaml.yaml
```

Comment:
I think `item` is undefined, shouldn't that still be `plugin`? I'm not sure that `.yaml.yaml` was intentional either.

Also, did this path really need to be updated to fix anything? I believe the following `extraVolumes` mount configuration is configured in the kubeadm-config files for the `admission-controls` hostPath directory, which mounted the plugin files in `{{ kube_config_dir }}/{{ plugin | lower }}.yaml` within the kube-apiserver container before this PR was merged:

```yaml
  - name: admission-control-configs
    hostPath: {{ kube_config_dir }}/admission-controls
    mountPath: {{ kube_config_dir }}
    readOnly: false
    pathType: DirectoryOrCreate
```

I think configuring `{{ kube_config_dir }}/admission-controls/` is a worthwhile change, but the extraVolumes entry would also need to be updated for this to work.

Reply:
Yes, you're perfectly right. I totally forgot that the control plane was in pods, when doing that, + the double extension is obviously wrong...

Should have added a reproducer first :grimacing:

Reply:
Thanks @chadswen